components/job-orchestration/job_orchestration/executor/query/extract_stream_task.py

```diff
@@ -146,3 +162,44 @@ def extract_stream(
         task_id=task_id,
         start_time=start_time,
     )
+
+    if enable_s3_upload and QueryTaskStatus.SUCCEEDED == task_results.status:
+        logger.info(f"Uploading streams to S3...")
+
+        upload_error = False
+        for line in task_stdout_as_str.splitlines():
+            try:
+                stream_stats = json.loads(line)
+            except json.decoder.JSONDecodeError:
+                logger.exception(f"`{line}` cannot be decoded as JSON")
+                upload_error = True
+                continue
+
+            stream_path_str = stream_stats.get("path")
+            if stream_path_str is None:
+                logger.error(f"`path` is not a valid key in `{line}`")
+                upload_error = True
+                continue
+
+            stream_path = Path(stream_path_str)
+
+            if not upload_error:
```

Comment:
```suggestion
            # If we've had a single upload error, we don't want to try uploading any other streams
            # since that may unnecessarily slow down the task and generate a lot of extraneous
            # output.
            if not upload_error:
```

Or something like that?

Reply:
Sure, I will take this (with fixing typos)